### PR TITLE
Image renaming to qa

### PR DESCRIPTION
**Pull request by armorcommander**
**Changes**
- Updated docker test images tags
  - Removed the python-3.x-<git_sha> tag since drone isn't going to be able to dynamically reference it
  - Added a tag ending in '-qa'
  - Added a tag ending in '-' to replicate what it was before the changes.
- Included the poetry install for python 3.9, 3.8, 3.7, 3.6 docker images

### CODE REVIEW

File: .github/workflows/docker_testing.yml [hunk 1]
```diff
@@ -100,19 +100,19 @@ jobs:
...
     - name: push
-      if: steps.buld_test.outcome == 'success'
+      if: steps.build_test.outcome == 'success'
       uses: elgohr/Publish-Docker-Github-Action@master
       with:
         name: 3mcloud/lambda-packager
         username: ${{ secrets.DOCKERHUBUSER }}
         password: ${{ secrets.DOCKERHUBPASS }}
-        tags: "node-12.16-${{ github.sha }}"
+        tags: "node-12.22-,node-12.22-qa"
```
Comment (erictwalker18) [inline .github/workflows/docker_testing.yml:113]:
```suggestion
        tags: "node-12.22-${{github.sha}},node-12.22-qa"
```

File: .github/workflows/docker_testing.yml [hunk 2]
```diff
@@ -121,16 +121,16 @@ jobs:
...
     - name: push
-      if: steps.buld_test.outcome == 'success'
+      if: steps.build_test.outcome == 'success'
       uses: elgohr/Publish-Docker-Github-Action@master
       with:
         name: 3mcloud/lambda-packager
         username: ${{ secrets.DOCKERHUBUSER }}
         password: ${{ secrets.DOCKERHUBPASS }}
-        tags: "node-14.18-${{ github.sha }}"
+        tags: "node-14.18-,node-14.18-qa"
```
Comment (erictwalker18) [inline .github/workflows/docker_testing.yml:134]:
```suggestion
        tags: "node-14.18--${{github.sha}},node-14.18-qa"
```

**Review by erictwalker18 — CHANGES_REQUESTED** [review]
The docker_testing and pull_request workflows are almost identical - I'd say entirely remove the pull_request one or make pull_request do the testing and docker_testing rename to `docker_push_qa` or similar and only do the docker push

**Comment by erictwalker18** [comment]
> The docker_testing and pull_request workflows are almost identical - I'd say entirely remove the pull_request one or make pull_request do the testing and docker_testing rename to `docker_push_qa` or similar and only do the docker push

I take it back, definitely do the latter, and have the `docker_push_qa` go for all branches, do not exclude master so we have the `image-name-${{github.sha}}` for every commit.

File: .github/workflows/docker_testing.yml [hunk 3]
```diff
@@ -16,17 +16,17 @@ jobs:
...
     - name: push
-      if: steps.buld_test.outcome == 'success'
+      if: steps.build_test.outcome == 'success'
       uses: elgohr/Publish-Docker-Github-Action@master
       with:
         name: 3mcloud/lambda-packager
         username: ${{ secrets.DOCKERHUBUSER }}
         password: ${{ secrets.DOCKERHUBPASS }}
-        tags: "python-3.6-${{github.sha}}"
+        tags: "python-3.6-,python-3.6-qa"
```
[the remark this answers is not in the document]
Comment (armorcommander) [reply .github/workflows/docker_testing.yml]: Removed the tag ending in '-' and added the tag ending with the commit sha

File: .github/workflows/docker_testing.yml [hunk 4]
```diff
@@ -37,17 +37,17 @@ jobs:
...
     - name: push
-      if: steps.buld_test.outcome == 'success'
+      if: steps.build_test.outcome == 'success'
       uses: elgohr/Publish-Docker-Github-Action@master
       with:
         name: 3mcloud/lambda-packager
         username: ${{ secrets.DOCKERHUBUSER }}
         password: ${{ secrets.DOCKERHUBPASS }}
-        tags: "python-3.7-${{ github.sha }}"
+        tags: "python-3.7-,python-3.7-qa"
```
[the remark this answers is not in the document]
Comment (armorcommander) [reply .github/workflows/docker_testing.yml]: Removed the tag ending in '-' and added the tag ending with the commit sha

File: .github/workflows/docker_testing.yml [hunk 5]
```diff
@@ -58,17 +58,17 @@ jobs:
...
     - name: push
-      if: steps.buld_test.outcome == 'success'
+      if: steps.build_test.outcome == 'success'
       uses: elgohr/Publish-Docker-Github-Action@master
       with:
         name: 3mcloud/lambda-packager
         username: ${{ secrets.DOCKERHUBUSER }}
         password: ${{ secrets.DOCKERHUBPASS }}
-        tags: "python-3.8-${{ github.sha }}"
+        tags: "python-3.8-,python-3.8-qa"
```
[the remark this answers is not in the document]
Comment (armorcommander) [reply .github/workflows/docker_testing.yml]: Removed the tag ending in '-' and added the tag ending with the commit sha

File: .github/workflows/docker_testing.yml [hunk 6]
```diff
@@ -79,17 +79,17 @@ jobs:
...
     - name: push
-      if: steps.buld_test.outcome == 'success'
+      if: steps.build_test.outcome == 'success'
       uses: elgohr/Publish-Docker-Github-Action@master
       with:
         name: 3mcloud/lambda-packager
         username: ${{ secrets.DOCKERHUBUSER }}
         password: ${{ secrets.DOCKERHUBPASS }}
-        tags: "python-3.9-${{ github.sha }}"
+        tags: "python-3.9-,python-3.9-qa"
```
[the remark this answers is not in the document]
Comment (armorcommander) [reply .github/workflows/docker_testing.yml]: Removed the tag ending in '-' and added the tag ending with the commit sha

File: .github/workflows/docker_testing.yml (hunk 1 above)
Comment (armorcommander) [reply .github/workflows/docker_testing.yml]: Removed the tag ending in '-' and added the tag ending with the commit sha

File: .github/workflows/docker_testing.yml (hunk 2 above)
Comment (armorcommander) [reply .github/workflows/docker_testing.yml]: Removed the tag ending in '-' and added the tag ending with the commit sha

**Review by erictwalker18 — APPROVED** [review]
LGTM

**Pull request merged**